src/components/LHNOptionsList/LHNOptionsList.js

```diff
@@ -0,0 +1,104 @@
+    contentContainerStyles: PropTypes.arrayOf(PropTypes.object).isRequired,
+
+    /** Sections for the section list */
+    data: PropTypes.arrayOf(PropTypes.string).isRequired,
+
+    /** Index for option to focus on */
+    focusedIndex: PropTypes.number.isRequired,
+
+    /** Callback to fire when a row is selected */
+    onSelectRow: PropTypes.func.isRequired,
+
+    /** Toggle between compact and default view of the option */
+    optionMode: PropTypes.oneOf(_.values(CONST.OPTION_MODE)).isRequired,
+
+    /** Callback to execute when the SectionList lays out */
+    onLayout: PropTypes.func.isRequired,
+};
+
+class LHNOptionsList extends Component {
+    constructor(props) {
+        super(props);
+
+        this.renderItem = this.renderItem.bind(this);
+        this.getItemLayout = this.getItemLayout.bind(this);
+    }
+
+    /**
+     * This function is used to compute the layout of any given item in our list. Since we know that each item will have the exact same height, this is a performance optimization
+     * so that the heights can be determined before the options are rendered. Otherwise, the heights are determined when each option is rendering and it causes a lot of overhead on large
+     * lists.
+     *
+     * @param {Array} data - This is the same as the data we pass into the component
+     * @param {Number} index the current item's index in the set of data
+     *
+     * @returns {Object}
+     */
+    getItemLayout(data, index) {
+        const optionHeight = this.props.optionMode === CONST.OPTION_MODE.COMPACT ? variables.optionRowHeightCompact : variables.optionRowHeight;
+        return {
+            length: optionHeight,
+            offset: index * optionHeight,
+            index,
+        };
+    }
+
+    /**
+     * Function which renders a row in the list
+     *
+     * @param {Object} params
+     * @param {Object} params.item
+     * @param {Number} params.index
+     *
+     * @return {Component}
+     */
+    renderItem({item, index}) {
+        return (
+            <OptionRowLHN
+                reportID={item}
+                viewMode={this.props.optionMode}
+                isFocused={this.props.focusedIndex === index}
+                onSelectRow={this.props.onSelectRow}
+            />
+        );
+    }
+
+    render() {
+        return (
+            <View style={[styles.flex1]}>
+                <FlatList
+                    indicatorStyle="white"
+                    keyboardShouldPersistTaps="always"
+                    contentContainerStyle={this.props.contentContainerStyles}
+                    showsVerticalScrollIndicator={false}
+                    data={this.props.data}
+                    keyExtractor={item => item}
+                    stickySectionHeadersEnabled={false}
+                    renderItem={this.renderItem}
+                    getItemLayout={this.getItemLayout}
+                    extraData={this.props.focusedIndex}
```

Comment:
I actually renamed that to `viewMode` and it's still there. It's what toggles between compact mode or not.

Reply:
I guess my question is whether it should it be in the `extraData` field as this is a way to [tell the component that it needs to update](https://reactnative.dev/docs/flatlist). Maybe it's not needed since changing the view mode should trigger a re-render of this component anyway.

Reply:
Ah yes, I don't think it's necessary to have in `extraData` because of exactly what you said.